.pre-commit-config.yaml

```diff
@@ -56,3 +56,32 @@ repos:
         name: Check import conventions; see CONTRIBUTING.rst#guidelines
         language: pygrep
         entry: "^ *import (networkx(,|$| (?!as nx))|numpy(?! as np)|scipy(?! as sp)|pandas(?! as pd)|matplotlib(,|$)|matplotlib.pyplot(?! as plt)|matplotlib (?!as mpl))"
+
+  # The hooks below do not run automatically (may be too slow, bothersome, or low value).
+  # They can be run manually such as before releasing a new version or annually. To run:
+  #
+  #   $ pre-commit run --all-files --hook-stage manual
+  #
+  # Automatically remove unused imports (but not as aggressively as ruff rule F401)
+  - repo: https://github.com/PyCQA/autoflake
+    rev: 0544741e2b4a22b472d9d93e37d4ea9153820bb1 # frozen: v2.3.1
+    hooks:
+      - id: autoflake
+        stages: [manual]
+        args: [--in-place]
```

Comment:
Not super familiar with precommit hooks or autoflake, so I may be missing something here, but should we add `-r` to make this run recursive? As best as I understand it, its only checking the top level files currently.